includes/ets_user_qa_function.php

```diff
@@ -194,29 +195,47 @@ public function ets_ask_qustion_tab() {
 			$productQaLength = get_option('ets_product_q_qa_list_length'); 
 			$loadMoreButton = get_option('ets_load_more_button'); 	
 			$pagingType = get_option('ets_product_qa_paging_type' ); 
-			$etsGetQuestion = get_post_meta( $productId,'ets_question_answer', true );
+			$all_questions = get_post_meta( $productId,'ets_question_answer', true );
+			
+			$etsGetQuestion = array_filter($all_questions, function ($value) {
+				if((isset($value['approve']) && $value['approve'] == 'yes') || !isset($value['approve'])){
+					if ($value) {
+						return $etsGetQuestion = $value;
```

Comment:
You are assigning and returning the value? what does it mean?
Read the [definition](https://www.php.net/manual/en/function.array-filter.php) again, observe the return type of the filter callback. Be a keen reader!

Reply:
its return filter array  and i am also return filter question array

Reply:
Wrong, the callback is not supposed to read the array.